Sources/core/objects/Movable.h

```diff
@@ -34,7 +34,7 @@ class IMovable : public IOnMapObject
     void processMove();//for move
     virtual bool TryMove(Dir direct);
     bool CheckMoveTime();
-    bool CheckPassable();
+    virtual bool CheckPassable();
     bool MainMove();
```

Comment:
Well I mean is it worth to make checkpassable virtual when it is only overloaded in projectiles,I mean we can make a ProjectileStatusCheck()(or name it as you wish) and do your idea(so it won't be deleted in process and I agree that's better because the delete can be called from an outside object).
```
bool Projectile::ProjectileStatusCheck()
{
    if (IMovable::CheckPassable())
    {
        return true;
    }
    Delete();
    return false;
}
```